features/networking/sdn.feature

```diff
@@ -300,3 +300,34 @@ Feature: SDN related networking scenarios
     Then the step should succeed
     And the output should contain "Hello OpenShift"
     """
+    
+  # @author anusaxen@redhat.com
+  # @case_id OCP-25787
+  @admin
+  Scenario: Don't write CNI configuration file until ovn-controller has done at least one iteration
+    Given the env is using "OVNKubernetes" networkType
+    And I select a random node's host
+    #Checking controller iteration 1
+    When I run command on the "<%= node.name %>" node's sdn pod:
+      | bash | -c | ovs-appctl -t ovn-controller connection-status |
+    Then the step should succeed
+    And the output should contain "connected"
+    #Checking controller iteration 2
+    When I run command on the "<%= node.name %>" node's sdn pod:
+      | bash | -c | ls -l /var/run/openvswitch/ |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/ovn-controller.\d*\.ctl/)[0]` is stored in the :controller_pid_file clipboard
+    #Checking controller iteration 3
+    When I run command on the "<%= node.name %>" node's sdn pod:
+      | bash | -c | ovs-appctl -t /var/run/openvswitch/<%= cb.controller_pid_file %> connection-status |
+    Then the step should succeed
+    And the output should contain "connected"
+    #Checking controller iteration 4
+    When I run command on the "<%= node.name %>" node's sdn pod:
+      | bash | -c | ovs-ofctl dump-flows br-int \| wc -l |
+    Then the step should succeed
+    And the expression should be true> @result[:response].match(/\d*/)[0].to_i> 0
+    #Checking final iteration post all above iterations passed. In this iteration we expect CNi file to be created
+    And I run commands on the host:
+      | ls -l /var/run/multus/cni/net.d/ |
+    And the output should contain "ovn-kubernetes.conf"
```

Comment:
After reading test case Scenario, the testing steps should be like break all four controller iterations, and checking "ls -l /var/run/multus/cni/net.d/" should fail, then make one of four controller iterations back to work, then checking "ls -l /var/run/multus/cni/net.d/" should pass. How do you think? @anuragthehatter

Reply:
@weliang1 this case is making sure that if all 4 steps shows success, then there should be CNI file written. If all steps shows success and no CNI is found then we know something is wrong or if CNI is written and one of step is down. These iterations only run when OVN cluster comes first time. I also had same thinking but as per talk with Casey few days back, its not easy to break all iterations (lot of iptables destructiveness involved and no gurantees) and should be avoided for cluster turmoil. Also CNI written is not dynamic, once its written, it will be there. Passing or Failing of this case will give a clue about healthy/unhealthy OVN cluster and iterations issue if any.

Reply:
Perhaps use
```
    And I run commands on the host:
      | ls /var/run/multus/cni/net.d/ovn-kubernetes.conf |
    Then the step should succeed
```

Reply:
Done and restested. Thanks for suggesting optimisation